tests/unit_tests/data_model/test_validate_data.py

```diff
@@ -554,16 +554,24 @@ def test_prepare_model_parameter():
     assert pytest.approx(data_validator.data_dict["value"][2]) == 3000.0
     assert data_validator.data_dict["unit"] == "km"
 
+    data_validator.data_dict["value"] = "1000. 2000. 3000."
     data_validator.data_dict["unit"] = "km, kg, s"
     data_validator._prepare_model_parameter()
     assert data_validator.data_dict["unit"][0] == "km"
     assert data_validator.data_dict["unit"][1] == "kg"
     assert data_validator.data_dict["unit"][2] == "s"
 
+    data_validator.data_dict["value"] = "1000. 2000. 3000."
     data_validator.data_dict["unit"] = ", , "
     data_validator._prepare_model_parameter()
     assert all(item == "" for item in data_validator.data_dict["unit"])
 
+    data_validator.data_dict["value"] = "1000 2000 3000"
+    data_validator.data_dict["unit"] = "ct"
+    data_validator.data_dict["type"] = "int64"
+    data_validator._prepare_model_parameter()
+    assert isinstance(data_validator.data_dict["value"][0], int)
+
```

Comment:
Maybe add a test for space separated units? I.e. 
`data_validator.data_dict["unit"] = "c t"` 
`assert data_validator.data_dict["unit"] == "ct"`

Reply:
Good point - added a test for this.

But I don't think your assumption is right: in astropy units `c t` would not resolve to `ct` but to `c*t`. So we should to the same thing. But note that this function is just separating the unit string; no other changes are applied.

Reply:
Okey true, thanks for adding the further tests. Looks all good now.